scripts/apps/search/components/item-list-popup.tsx

```diff
@@ -0,0 +1,60 @@
+import React, {ElementType, CSSProperties, SyntheticEvent} from 'react';
+import {createPortal} from 'react-dom';
+import {positionPopup} from '../helpers';
+
+interface IProps {
+    label: string;
+    close: () => void;
+    target: ElementType | EventTarget;
+}
+
+const BACKDROP_STYLE: CSSProperties = {
+    position: 'absolute',
+    top: 0,
+    bottom: 0,
+    left: 0,
+    right: 0,
+    zIndex: 999, // positionPopup sets it to 1000
+    overflow: 'auto', // to allow scroll
+};
+
+export class ItemListPopup extends React.PureComponent<IProps> {
+    constructor(props) {
+        super(props);
+        this.close = this.close.bind(this);
+    }
+
+    componentDidMount() {
+        positionPopup(this.props.target);
+    }
+
+    close(event: SyntheticEvent) {
+        event.stopPropagation();
```

Comment:
Why do you need this?

Reply:
to avoid the click propagate to list item and open preview

Reply:
The popup is outside of the list component it should not send it any events anyhow. It's probably because you are using a portal. Can you comment that so it's clear for future readers?

Reply:
yep that's how portals work

Reply:
That's why we shouldn't be using a portal here :) Components like modals/dropdowns/popups should be isolated and not forward any events to where they have been initiated from. We can sprinkle `preventDefault`s and `stopPropagation` calls, but that's a sub-optimal solution.

Reply:
I'm not sure what's better actually, there must be some kind of connection because if that parent component (item) is not rendered it shouldn't be visible anymore.

Reply:
I agree, we need a connection for removing the popup when the parent is umounted, but we can do it through a wrapper component :
```
class Wrapper extends React.PureComponent {
    componentDidMount() {
        // render to DOM
    }
    componentDidUnmount() {
        // clean up DOM
    }
    render() {
        return null;
    }
}
```

Reply:
yep, or we use portal

Reply:
What I like about the wrapper approach is that it gives confidence that bugs similar to this one are less likely.

Here's one present right now: Let's say I have a popup open and want to highlight a word to be copied. I use the browser feature to double click the word in order to highlight it and article edit view opens :beetle:

Reply:
I'm not sure it's a bug or a feature. This wrapper is a workaround to avoid this click propagate and imo that's less explicit than adding click handler inside the component which would stop the propagation.

Reply:
It sounds like you want it to be a feature so you don't have to change the code :smile:

The current code is fine for this use case. There is a small bug with the double click, but it's fine. I expect to replace this with a generic popup component as soon as we get it in the UI framework.

Being explicit is fine, but I don't see this being about explicit/implicit code. It's more about building an abstraction and isolating it. When React calls `document.createElement` you don't see it being too implicit and that you'd rather call it yourself. You can also look at a native select dropdown. If you double click an option it won't send the event to the parent where that select is rendered in. I see no uses cases where it would make sense.